### PR TITLE
Added pytest-extra-durations to monitor the speed of the tests.

### LINKED ISSUE #1316: Monitoring the speed of the tests.
**Describe the feature and the current behavior/state.**

The CI should fail when the tests are not fast enough to prevent merging tests that are too time consuming. 

Technically, with pytest it's easy, there are even pugins to add timeouts. But we cannot take this implementation.

The problem is that test times are not deterministic. They vary from one run to another. So merging a pull request with the CI passing (and timeout implemented) does not garantee that the CI will continue to pass in the future.

Ideally, we should have the tests times of the first commit of the pull request branch and compare that to the new test times in the pull request, and make the CI fail if we notice that some tests have been changed/added and the running time for those tests increased too much. So we should put a limit on the difference of the running times, not the test times themselves. No idea how to implement that though.

**Relevant information**
- Are you willing to contribute it (yes/no): yes
- Are you willing to maintain it going forward? (yes/no): yes
- Is there a relevant academic paper? (if so, where): no
- Is there already an implementation in another framework? (if so, where): I wish I knew. Does anyone know an open source project where the test times are monitored?
- Was it part of tf.contrib? (if so, where): no

**Which API type would this fall under (layer, metric, optimizer, etc.)**

testing, no API

**Who will benefit with this feature?**

The reviewers, they wouldn't have to check the running times of the tests added by the contributors.

**Any other info.**

nope

### PR DESCRIPTION
**Pull request by gabrieldemarmiesse**
Should be enough to ensure the duration of the tests doesn't blow up out of nowhere.

Closes #1316

Current output:

```
========================== slowest modules durations ===========================
49.85s tensorflow_addons/seq2seq/tests/attention_wrapper_test.py
24.94s tensorflow_addons/layers/tests/normalizations_test.py
17.38s tensorflow_addons/image/tests/filters_test.py
14.21s tensorflow_addons/optimizers/tests/lookahead_test.py
14.14s tensorflow_addons/optimizers/tests/rectified_adam_test.py
======================= slowest test functions durations =======================
11.66s tensorflow_addons/layers/tests/optical_flow_test.py::test_gradients
10.53s tensorflow_addons/optimizers/tests/cyclical_learning_rate_test.py::test_exponential_cyclical_learning_rate
7.97s tensorflow_addons/optimizers/tests/lookahead_test.py::test_fit_simple_linear_model
7.69s tensorflow_addons/optimizers/tests/novograd_test.py::test_fit_simple_linear_model
6.88s tensorflow_addons/optimizers/tests/stochastic_weight_averaging_test.py::test_fit_simple_linear_model
6.00s tensorflow_addons/layers/tests/normalizations_test.py::test_groupnorm_2d_different_groups
5.97s tensorflow_addons/optimizers/tests/moving_average_test.py::test_fit_simple_linear_model
5.58s tensorflow_addons/seq2seq/tests/attention_wrapper_test.py::AttentionMechanismTest::test_save_load_layer_bahdanau_monotonic
5.28s tensorflow_addons/seq2seq/tests/attention_wrapper_test.py::AttentionMechanismTest::test_save_load_layer_bahdanau
5.18s tensorflow_addons/seq2seq/tests/attention_wrapper_test.py::AttentionMechanismTest::test_save_load_layer_luong
5.04s tensorflow_addons/seq2seq/tests/attention_wrapper_test.py::AttentionMechanismTest::test_save_load_layer_luong_monotonic
3.81s tensorflow_addons/optimizers/tests/lookahead_test.py::test_sparse_exact_ratio
3.64s tensorflow_addons/layers/tests/normalizations_test.py::test_picture_input
3.43s tensorflow_addons/layers/tests/wrappers_test.py::test_model_build
3.31s tensorflow_addons/layers/tests/netvlad_test.py::test_simple
2.97s tensorflow_addons/seq2seq/tests/attention_wrapper_test.py::AttentionWrapperTest::testLuongScaled
2.94s tensorflow_addons/image/tests/dense_image_warp_test.py::test_interpolation
2.76s tensorflow_addons/callbacks/tests/time_stopping_test.py::test_time_stopping_verbose
2.69s tensorflow_addons/losses/tests/npairs_test.py::test_unweighted
2.67s tensorflow_addons/seq2seq/tests/attention_wrapper_test.py::AttentionMechanismTest::test_memory_re_setup_bahdanau_monotonic
========================== sum of all tests durations ==========================
271.92s
```

### CODE REVIEW

**Comment by AakashKumarNain** [comment]
LGTM! Do you want to make any other changes or should I merge this one now? @gabrieldemarmiesse

**Comment by gabrieldemarmiesse** [comment]
That's all folks!

**Comment by AakashKumarNain** [comment]
Thanks
